rayon-core/src/thread_pool/mod.rs

```diff
@@ -97,24 +119,46 @@ impl ThreadPool {
         }
     }
 
-    /// Spawns an asynchronous task in this thread-pool. See
-    /// `spawn_async()` for more details.
+    /// Execute `oper_a` and `oper_b` in the thread-pool and return
+    /// the results. Equivalent to `self.install(|| join(oper_a,
+    /// oper_b))`.
     #[cfg(feature = "unstable")]
-    pub fn spawn_async<OP>(&self, op: OP)
+    pub fn join<A, B, RA, RB>(&self, oper_a: A, oper_b: B) -> (RA, RB)
+        where A: FnOnce() -> RA + Send,
+              B: FnOnce() -> RB + Send,
+              RA: Send,
+              RB: Send
+    {
+        self.install(|| join(oper_a, oper_b))
+    }
+
+    /// Execute `oper_a` and `oper_b` in the thread-pool and return
+    /// the results. Equivalent to `self.install(|| scope(...))`.
+    #[cfg(feature = "unstable")]
+    pub fn scope<'scope, OP, R>(&self, op: OP) -> R
+        where OP: for<'s> FnOnce(&'s Scope<'scope>) -> R + 'scope + Send, R: Send
+    {
+        self.install(|| scope(op))
+    }
+
+    /// Spawns an asynchronous task in this thread-pool. See `spawn()`
+    /// for more details.
+    #[cfg(feature = "unstable")]
+    pub fn spawn<OP>(&self, op: OP)
```

Comment:
These docs should be more clear what `spawn()` it's referencing - presumably the global one and not a recursive reference.  Maybe even put a link on that.